### PR TITLE
Add email activation

### PR DESCRIPTION
**Pull request by leojiaxingli**

### CODE REVIEW

**Comment by junhaoliao** [comment]
@leojiaxingli Thanks for your contribution!
The TTL cache approach is impressive, and I personally learned a few great lessons from your approaches.

Could you check whether the following changes will break your code:
1. a few variable names: 'resend_cache' -> 'resend_cooldown'; 'RESEND_TTL_SECOND' -> 'RESEND_COOLDOWN_TTL_SECOND'
2. the method signature of "send_activation_email" and also the parameters in api('/resend') because the login fields only contains usernames. (Email login support is currently removed. @Kruz-X )
3. use env variables to read sender email credentials
4. replace the sender to use support@ictrl.ca hosted by Tencent Work Email. I do observe some delays in receiving the emails.

If there is no objection, I will probably replace our email server with Zoho:
1. It is the current host of junhao@junhao.ca; I can tell it is fairly reliable.
2. It does provide SMTP
3. It doesn't provide free IMAP/POP3: does it matter to us? @leojiaxingli 

@Kruz-X There are some significant changes in the Login frontend. Would you mind familiarizing yourself with the changes and add some real-time password check in the frontend? The password requirements are available in https://github.com/junhaoliao/iCtrl/blob/2f388ae223fcd7bee236e693c3c504ea2bf1c2d2/application/utils.py#L39

@KevinHaoranZhang Could you verify if the changes break our existing code on both the frontend and backend?

File: application/resources/activation_email_template.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <p>
+        Best Regards,</p>
+    <p>
+        <b>iCtrl Support</b><br/>
+        <a href="mailto:support@ictrl.ca">support@ictrl.ca</a>
+    </p>
+
+</div>
+</body>
+</html>
```
Comment (KevinHaoranZhang) [inline application/resources/activation_email_template.html:26]: Add new line

File: client/package.json [hunk 2]
```diff
@@ -52,4 +53,4 @@
   },
   "proxy": "http://localhost:5000",
   "//": "https://ictrl.ca"
-}
+}
```
Comment (KevinHaoranZhang) [inline client/package.json:56]: Are JSON files required for future updates? If not, please update gitignore. 
Also needs a new line character

**Review by KevinHaoranZhang — CHANGES_REQUESTED** [review]
Verified email activation design. The frontend user interface and backend profiling separation are aligned with previously outlined design requirements. Please fix minor syntax errors before approval.

File: client/package.json (hunk 2 above)
Comment (junhaoliao) [reply client/package.json:56]: package.json specifies the dependencies and running configurations of our project. I think they do need to be in the index.

**Pull request merged**